dask/array/core.py

```diff
@@ -1176,7 +1178,7 @@ def from_func(func, shape, dtype=None, name=None, args=(), kwargs={}):
     """
     if args or kwargs:
         func = partial(func, *args, **kwargs)
-    name = name or next(names)
+    name = name or 'from_func-' + tokenize((func, shape, dtype, args, kwargs))
```

Comment:
I was also thinking about that. Not sure how many layers down we'd want to normalize though. Just the first level of the tuple? I'm hesitant to do a full normalization of a whole nested structure, so probably something simple like that, and expose the `normalize` function as well for external use.